packages/amplify-category-storage/src/provider-utils/awscloudformation/service-walkthroughs/s3-user-input-state.ts

```diff
@@ -0,0 +1,162 @@
+  dependsOn: S3CFNDependsOn[];
+};
+
+export type S3InputStateOptions = {
+  userInputDataFilename: string; //cli-inputs.json
+  resourceName: string;
+  inputPayload?: S3UserInputs;
+};
+
+export class S3InputState {
+  static s3InputState: S3InputState;
+  _cliInputsFilePath: string; //cli-inputs.json (output) filepath
+  _resourceName: string; //user friendly name provided by user
+  _category: string; //category of the resource
+  _service: string; //AWS service for the resource
+  _inputPayload: S3UserInputs | undefined; //S3 options selected by user
+
+  constructor(props: S3InputStateOptions) {
+    this._category = AmplifyCategories.STORAGE;
+    this._service = AmplifySupportedService.S3;
+    this._cliInputsFilePath = props.userInputDataFilename;
+    this._resourceName = props.resourceName;
+
+    // Read cliInputs file if exists
+    try {
+      this._inputPayload = props.inputPayload ?? JSONUtilities.readJson(this._cliInputsFilePath, { throwIfNotExist: true });
+    } catch (e) {
+      throw new Error('migrate project with command : amplify migrate <to be decided>');
+    }
+
+    // validate cli-inputs.json
+    const schemaValidator = new CLIInputSchemaValidator(this._service, this._category, 'S3UserInputs');
+    schemaValidator.validateInput(JSON.stringify(this._inputPayload!));
+  }
+
+  public static getPermissionTypeFromCfnType(s3pPrmissionCfnType: S3CFNPermissionType): S3PermissionType {
+    switch (s3pPrmissionCfnType) {
+      case S3CFNPermissionType.PUT_OBJECT:
+        return S3PermissionType.PUT_OBJECT;
+      case S3CFNPermissionType.GET_OBJECT:
+        return S3PermissionType.GET_OBJECT;
+      case S3CFNPermissionType.DELETE_OBJECT:
+        return S3PermissionType.DELETE_OBJECT;
+      case S3CFNPermissionType.LIST_BUCKET:
+        return S3PermissionType.LIST_BUCKET;
+      default:
+        throw new Error(`Unknown CFN Type: ${s3pPrmissionCfnType}`);
+    }
+  }
+
+  public static getInputPermissionsFromCfnPermissions(selectedGuestPermissions: S3CFNPermissionType[] | undefined) {
+    if (selectedGuestPermissions) {
+      return selectedGuestPermissions.map(S3InputState.getPermissionTypeFromCfnType);
+    } else {
+      return [];
+    }
+  }
+
+  public static cliWalkThroughToCliInputParams(cliInputsFilePath: string, options: S3CLIWalkthroughParams) {
+    const inputProps: S3InputStateOptions = {
+      userInputDataFilename: cliInputsFilePath,
+      resourceName: options.resourceName,
+      inputPayload: {
+        resourceName: options.resourceName,
+        bucketName: options.bucketName,
+        storageAccess: options.storageAccess,
+        selectedGuestPermissions: S3InputState.getInputPermissionsFromCfnPermissions(options.selectedGuestPermissions),
+        selectedAuthenticatedPermissions: S3InputState.getInputPermissionsFromCfnPermissions(options.selectedAuthenticatedPermissions),
+        isTriggerEnabled: options.triggerFunction !== 'NONE' ? true : false, //enable if trigger
```

Comment:
```suggestion
        isTriggerEnabled: options.triggerFunction !== 'NONE', //enable if trigger
```

Reply:
This file is a placeholder and would be overridden by @sachscode PR tied to S3 overrides.

Reply:
Same as above. This key will be going away in @sachscode's PR for S3.